community/io/src/test/java/org/neo4j/adversaries/pagecache/AdversarialPageCursor.java

```diff
@@ -0,0 +1,280 @@
+    }
+
+    @Override
+    public void putInt( int value )
+    {
+        adversary.injectFailure( SecurityException.class );
+        delegate.putInt( value );
+    }
+
+    @Override
+    public void putInt( int offset, int value )
+    {
+        adversary.injectFailure( SecurityException.class );
+        delegate.putInt( offset, value );
+    }
+
+    @Override
+    public long getUnsignedInt()
+    {
+        return currentReadIsInconsistent ? 0 : delegate.getUnsignedInt();
+    }
+
+    @Override
+    public long getUnsignedInt( int offset )
+    {
+        return currentReadIsInconsistent ? 0 : delegate.getUnsignedInt( offset );
+    }
+
+    @Override
+    public void getBytes( byte[] data )
+    {
+        if ( !currentReadIsInconsistent )
+        {
+            delegate.getBytes( data );
+        }
+    }
+
+    @Override
+    public void getBytes( byte[] data, int arrayOffset, int length )
+    {
+        if ( !currentReadIsInconsistent )
+        {
+            delegate.getBytes( data, arrayOffset, length );
+        }
+    }
+
+    @Override
+    public void putBytes( byte[] data )
+    {
+        adversary.injectFailure( SecurityException.class );
+        delegate.putBytes( data );
+    }
+
+    @Override
+    public void putBytes( byte[] data, int arrayOffset, int length )
+    {
+        adversary.injectFailure( SecurityException.class );
+        delegate.putBytes( data, arrayOffset, length );
+    }
+
+    @Override
+    public short getShort()
+    {
+        return currentReadIsInconsistent ? 0 : delegate.getShort();
+    }
+
+    @Override
+    public short getShort( int offset )
+    {
+        return currentReadIsInconsistent ? 0 : delegate.getShort( offset );
+    }
+
+    @Override
+    public void putShort( short value )
+    {
+        adversary.injectFailure( SecurityException.class );
+        delegate.putShort( value );
+    }
+
+    @Override
+    public void putShort( int offset, short value )
+    {
+        adversary.injectFailure( SecurityException.class );
+        delegate.putShort( offset, value );
+    }
+
+    @Override
+    public void setOffset( int offset )
+    {
+        adversary.injectFailure( SecurityException.class );
+        delegate.setOffset( offset );
+    }
+
+    @Override
+    public int getOffset()
+    {
+        adversary.injectFailure( SecurityException.class );
+        return delegate.getOffset();
+    }
+
+    @Override
+    public long getCurrentPageId()
+    {
+        adversary.injectFailure( SecurityException.class );
+        return delegate.getCurrentPageId();
+    }
+
+    @Override
+    public int getCurrentPageSize()
+    {
+        adversary.injectFailure( SecurityException.class );
+        return delegate.getCurrentPageSize();
+    }
+
+    @Override
+    public File getCurrentFile()
+    {
+        adversary.injectFailure( SecurityException.class );
+        return delegate.getCurrentFile();
+    }
```

Comment:
`getOffset`, `getCurrentPageId`, `getCurrentPageSize` and `getCurrentFile` don't throw anything.